src/data/data1d/Spectrum1D/peaks/optimizePeaks.ts

```diff
@@ -0,0 +1,29 @@
+import { PeakXYWidth } from 'cheminfo-types';
+import { xFindClosestIndex } from 'ml-spectra-processing';
+import { xyPeaksOptimization } from 'nmr-processing';
+
+import { Datum1D, Peak } from '../../../types/data1d';
+
+import { mapPeaks } from './mapPeaks';
+
+interface OptimizePeaksOptions {
+  from: number;
+  to: number;
+}
+
+export function optimizePeaks(datum1D: Datum1D, options: OptimizePeaksOptions) {
+  const { from, to } = options;
+  let { re, x } = datum1D.data;
+
+  if (from !== undefined && to !== undefined) {
```

Comment:
How could from or to be undefined if in the type definition they are mandatory and numbers ?

Reply:
we need it in one case when we full zoom out, but in any way this case can not happen because I always pass the window range